pkg/operator/ceph/object/controller.go

```diff
@@ -237,6 +237,10 @@ func (r *ReconcileCephObjectStore) reconcile(request reconcile.Request) (reconci
 		if err != nil {
 			return reconcile.Result{}, *cephObjectStore, errors.Wrapf(err, "failed to get object context")
 		}
+
+		// The object store context needs the CephCluster spec to read networkinfo
+		// Otherwise GetAdminOPSUserCredentials() will fail detecting the network provider when running RunAdminCommandNoMultisite()
+		objCtx.CephClusterSpec = cephCluster.Spec
```

Comment:
Why not modify the `NewMultisiteContext()` function that creates the context to fill in this info rather than doing it after the fact?

Reply:
Yaa, I thought that too, but it may require more investigation into all the controllers in the object pkg. This is the first version of the PR so that I won't miss this issue